src/ui/toolbar_block.h

```diff
@@ -0,0 +1,250 @@
+    }
+
+private:
+    template<play_stop::State buttonType>
+    inline void handleButton() {
+        using namespace gr::message;
+
+        constexpr static auto buttonName = [] {
+            switch (buttonType) {
+            case PlayStop:
+                return "\uf051";
+            case Play:
+                return "\uf04b";
+            case PlayStream:
+                return "\uf04e";
+            case Pause:
+                return "\uf04c";
+            case Stopped:
+                return "\uf04d";
+            case Error:
+            default:
+                return "Error";
+            }
+        };
+        const float actualButtonSize = 28.f;
+        const bool  disabled         = this->isStateDisabled(buttonType);
+        ImGui::BeginDisabled(disabled);
+        ImGui::PushFont(DigitizerUi::App::instance().fontIconsSolid);
+        ImGui::PushStyleVar(ImGuiStyleVar_FrameRounding, .5f * actualButtonSize);
+        const bool clicked = ImGui::Button(buttonName(), ImVec2(actualButtonSize, actualButtonSize));
+        ImGui::PopFont();
+        ImGui::SameLine();
+        ImGui::PopStyleVar();
+        ImGui::EndDisabled();
+        if (clicked && !disabled) {
+            if (auto e = this->changeToolStateTo(buttonType); e) {
+                this->emitMessage(this->ctrlOut, { { key::Kind, kind::SettingsChanged },
+                                                         { key::What, fmt::format("{} pressed", magic_enum::enum_name(buttonType)) } });
+            } else {
+                this->emitMessage(this->msgOut, { { key::Kind, kind::Error },
+                                                        { key::ErrorInfo, e.error().message },
+                                                        { key::Location, e.error().srcLoc() } });
+            }
+        }
+    }
+};
+
+template<typename T>
+struct LabelToolbarBlock
+    : public gr::Block<LabelToolbarBlock<T>, gr::BlockingIO<false>, gr::Drawable<gr::UICategory::Toolbar, "Dear ImGui">> {
+    gr::MsgPortIn ctrlIn;
+    std::string   message = "<no message>";
+
+    void
+    processMessages(auto &, std::span<const gr::Message> messages) {
+        using namespace gr::message;
+        for (const gr::Message &msg : messages) {
+            if (msg.contains(key::Kind) && msg.contains(key::What) && std::get<std::string>(msg.at(key::Kind)) == kind::SettingsChanged) {
+                this->settings().set({ { std::string("message"), std::get<std::string>(msg.at(key::What)) } });
+            }
+        }
+    }
+
+    gr::work::Status
+    draw() noexcept {
+        this->processScheduledMessages();
+        std::ignore                   = this->settings().applyStagedParameters(); // return ignored since there are no tags to be forwarded
+        const gr::work::Status status = gr::work::Status::OK;                     // this->invokeWork(); // calls work(...) -> processOne(...) (all in the same thread as this 'draw()'
+        ImGui::Text(message.c_str());
```

Comment:
Just FYI: this breaks with stricter warning settings(which somehow are the default on my machine) since the format string is not a constant  -> use either `ImGui::TextUnformatted(...)`  or `ImGui::Text("%s", ...)`